app/lib/user/delete_account_dialog.dart

```diff
@@ -0,0 +1,148 @@
+import 'package:air/user/user.dart';
+import 'package:logging/logging.dart';
+import 'package:provider/provider.dart';
+import 'package:iconoir_flutter/iconoir_flutter.dart' as iconoir;
+
+const _confirmationText = 'delete';
+
+final _log = Logger("DeleteAccountDialog");
+
+class DeleteAccountDialog extends HookWidget {
+  const DeleteAccountDialog({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final isConfirmed = useState(false);
+
+    final controller = useTextEditingController();
+
+    final colors = CustomColorScheme.of(context);
+    final loc = AppLocalizations.of(context);
+
+    return AirDialog(
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Center(
+            child: Text(
+              loc.deleteAccountScreen_title,
+              style: TextStyle(
+                fontSize: HeaderFontSize.h4.size,
+                fontWeight: FontWeight.bold,
+              ),
+            ),
+          ),
+          const SizedBox(height: Spacings.m),
+
+          Center(
+            child: iconoir.WarningCircle(
+              width: 40,
+              height: 40,
+              color: colors.function.danger,
+            ),
+          ),
+
+          const SizedBox(height: Spacings.s),
+
+          FieldLabel(loc.deleteAccountScreen_explanatoryText),
+
+          const SizedBox(height: Spacings.xs),
+
+          TextFormField(
+            autocorrect: false,
+            autofocus: true,
+            controller: controller,
+            decoration: airInputDecoration.copyWith(
+              hintText: loc.deleteAccountScreen_confirmationInputHint,
+              filled: true,
+              fillColor: colors.backgroundBase.secondary,
+            ),
+            onChanged: (value) =>
+                isConfirmed.value = value == _confirmationText,
+          ),
+
+          const SizedBox(height: Spacings.xs),
+
+          FieldLabel(loc.deleteAccountScreen_confirmationInputLabel),
+
+          const SizedBox(height: Spacings.m),
+
+          Row(
+            children: [
+              Expanded(
+                child: TextButton(
+                  onPressed: () {
+                    Navigator.of(context).pop(false);
+                  },
+                  style: airDialogButtonStyle.copyWith(
+                    backgroundColor: WidgetStatePropertyAll(
+                      colors.accent.quaternary,
+                    ),
+                  ),
+                  child: Text(
+                    loc.editDisplayNameScreen_cancel,
+                    style: TextStyle(fontSize: LabelFontSize.base.size),
+                  ),
+                ),
+              ),
+
+              const SizedBox(width: Spacings.xs),
+
+              Expanded(
+                child: AirDialogProgressTextButton(
+                  onPressed: (inProgress) =>
+                      _deleteAccount(context, inProgress),
```

Comment:
It's possible to tap the delete button even when 'delete' is not entered (found out the hard way)